### DISCUSSION TITLE
How do measure the amount of time spent in the last arm of a Y-maze?

### DISCUSSION THREAD

**Discussion by Goncalomelo95** [bonsai-rx/bonsai#1922] [Q&A]
In the attachment, it is my Bonsai flow. When the video stops, the info in the last arm can't be saved; the information is only saved when the animal changes arms. How can I handle this? Which noodles should I use to compile the last data?

Thank you in advance

![image](https://github.com/user-attachments/assets/d3a8903f-7ef1-4626-b83d-8d3effaf2284)

**Comment by bruno-f-cruz** [comment]
Hi,
Please share the code to the workflow. It is really hard to get help to complex workflows if people do not have access to the full code.

My best guess at this point is probably because one of the zip branches does not emit a match value or your skip+zip logic is not accounting for the very last trial (Since you need to match trial N with trial N + 1).
